src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipeline.scala

```diff
@@ -102,17 +102,29 @@ object ExtractionPipeline {
       linkedField = "library.accession"
     )
 
+    //[partition] that stream to separate [type=experiment]
+    // from [type=functional-characterization-experiment]
+    val (expReplicate, fcReplicate) = replicates.partition {
+      replicate => replicate("experiment").exists(experimentString => experimentString.equals("Experiment"))
```

Comment:
I think this is off. `experimentString` has type `Json`, so `equals` will always return `false` when the RHS is a `String`. The values of the `"experiment"` are also going to be IDs, so I don't think we can check against a fixed constant.

I'm reconsidering the anonymous function usage. Since this is just an optimization it's tough to see if this logic is working based on pipeline outputs, and it's tough to unit test when it's embedded inside the `partition` call. I'd split out a new method:
```scala
def isFunctionalCharacterizationReplicate(replicate: JsonObject): Boolean = ???
```
We can unit-test the method using some cherry-picked examples pulled from ENCODE.

Reply:
Added a unit test!